src/ng/filter/filter.js

```diff
@@ -127,7 +127,7 @@
  */
 function filterFilter() {
   return function(array, expression, comparator) {
-    if (!isArray(array)) {
+    if (!isArrayLike(array)) {
```

Comment:
This is not sufficient. We are using `array.filter`, which `isArrayLike` does not guarrantee to exist.

Maybe we could change `array.filter(...)` with something like `isArray(array) ? array.filter(...) : Array.prototype.filter.call(array, ...)`.
It might even be better to always use the latter form regardless of the type of `array`.

This needs a little more investigation.